### ISSUE TITLE
Homebridge Service stoped after Update 3.2.6

### ISSUE THREAD

**Issue by xalex88** [hoobs-org/HOOBS#649]
**Description**

**Version**
3.2.6

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ x] Yes
- [ ] No

**Pervious version**
3.2.4

**Did you orginally upgrade to HOOBS 3 
- [x] Yes
- [ ] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [ ] HOOBS SD Card
- [ x] Custom Install

**If this is a custom install, what's your operating system?**
MacOS

**List your plugins**
Please include a list of the plugins you are using. We need to be able to find it on NPM. If the plugin is no published, please include a link to the repository.
Only the Shelly Plugin

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
POST CONFIG HERE
```

**Post your log**
You can get the log from the intercafe. This is in the Log section.

```

23.5.2020, 19:15:14 HOOBS listening on port 8080.
23.5.2020, 19:15:16 Loaded plugin "homebridge-shelly".
23.5.2020, 19:15:18 [Shelly] 10 accessories loaded from cache
23.5.2020, 19:15:18 Bridge is running on port 51826.
23.5.2020, 19:15:18 [Shelly] Admin server is running on port 8081
23.5.2020, 19:15:19 Error: Invalid IV length
    at Cipheriv.createCipherBase (internal/crypto/cipher.js:79:18)
    at Cipheriv.createCipherWithIV (internal/crypto/cipher.js:115:20)
    at new Cipheriv (internal/crypto/cipher.js:217:22)
    at Object.createCipheriv (crypto.js:109:10)
    at Object.chacha20_poly1305_encryptAndSeal (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/util/hapCrypto.js:79:35)
    at HAPServer._this._handlePairVerifyStepOne (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:486:39)
    at HAPServer._this._handlePairVerify (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:450:23)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:227:39)
    at IncomingMessage.emit (events.js:198:13)
    at endReadableNT (_stream_readable.js:1129:12)
23.5.2020, 19:15:20 Error: Invalid IV length
    at Cipheriv.createCipherBase (internal/crypto/cipher.js:79:18)
    at Cipheriv.createCipherWithIV (internal/crypto/cipher.js:115:20)
    at new Cipheriv (internal/crypto/cipher.js:217:22)
    at Object.createCipheriv (crypto.js:109:10)
    at Object.chacha20_poly1305_encryptAndSeal (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/util/hapCrypto.js:79:35)
    at HAPServer._this._handlePairVerifyStepOne (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:486:39)
    at HAPServer._this._handlePairVerify (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:450:23)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:227:39)
    at IncomingMessage.emit (events.js:198:13)
    at endReadableNT (_stream_readable.js:1129:12)
23.5.2020, 19:15:20 Got SIGTERM, shutting down Bridge...

```

**Additional context**

**Comment by galzil** [comment]
Same here

**Comment by askovi** [comment]
This could be related to your nodejs version. Can you run this in Terminal and post the results:

`node -v`

**Comment by galzil** [comment]
this is the version: 
`v10.16.3`

**Comment by askovi** [comment]
Run these commands in terminal to update node and see if it fixes the issue:


`wget -q -O - http://bit.ly/get-hoobs | sudo bash /dev/stdin —node 12.16.3`

Then

`sudo reboot`

**Comment by galzil** [comment]
Just did:
`sudo npm install -g n`
and than
`sudo n stable`
did the job

**Comment by xalex88** [comment]
I did it like this and node was updated.  Now it hangs in the boot loop
![0C9CDE54-0EEC-44AD-9944-F279E837C7EE](https://user-images.githubusercontent.com/19264417/82746239-e2d85780-9d8d-11ea-945a-142a0853b1ed.png)

**Comment by askovi** [comment]
Login via ssh with user hoobs and password hoobsadmin then run this command to get the logs:

`sudo hoobs service log`

**Comment by Slartybart** [comment]
I've hit the same issue:  updated node but not the machine gets stuck in a boot loop like xalex88 above.  Here's my output:
Started HOOBS.
innerError Error: Cannot find module '../build/Debug/pty.node'
Require stack:
- /usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js
- /usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/index.js
- /usr/local/lib/node_modules/@hoobs/hoobs/server/api.js
- /usr/local/lib/node_modules/@hoobs/hoobs/server/cli.js
- /usr/local/lib/node_modules/@hoobs/hoobs/scripts/loader.js
- /usr/local/lib/node_modules/@hoobs/hoobs/bin/hoobs
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js:30:15)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/index.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/server/api.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/server/cli.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/scripts/loader.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/bin/hoobs'
  ]
}
/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js:35
        throw outerError;
        ^
Error: The module '/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/build/Release/pty.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 64. This version of Node.js requires
NODE_MODULE_VERSION 72. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1183:18)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js:26:11)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
hoobs.service: Main process exited, code=exited, status=1/FAILURE
hoobs.service: Failed with result 'exit-code'.
hoobs.service: Service RestartSec=3s expired, scheduling restart.
hoobs.service: Scheduled restart job, restart counter is at 13.
Stopped HOOBS.
Started HOOBS.
innerError Error: Cannot find module '../build/Debug/pty.node'
Require stack:
- /usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js
- /usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/index.js
- /usr/local/lib/node_modules/@hoobs/hoobs/server/api.js
- /usr/local/lib/node_modules/@hoobs/hoobs/server/cli.js
- /usr/local/lib/node_modules/@hoobs/hoobs/scripts/loader.js
- /usr/local/lib/node_modules/@hoobs/hoobs/bin/hoobs
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js:30:15)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/index.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/server/api.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/server/cli.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/scripts/loader.js',
    '/usr/local/lib/node_modules/@hoobs/hoobs/bin/hoobs'
  ]
}
/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js:35
        throw outerError;
        ^
Error: The module '/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/build/Release/pty.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 64. This version of Node.js requires
NODE_MODULE_VERSION 72. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1183:18)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/node-pty/lib/unixTerminal.js:26:11)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
hoobs.service: Main process exited, code=exited, status=1/FAILURE
hoobs.service: Failed with result 'exit-code'.
hoobs.service: Service RestartSec=3s expired, scheduling restart.
hoobs.service: Scheduled restart job, restart counter is at 14.
Stopped HOOBS.

**Comment by Slartybart** [comment]
OK... I followed some old advice from the facebook page which seems to have worked:

sudo npm uninstall -g @hoobs/hoobs
sudo npm install -g --unsafe-perm @hoobs/hoobs
// reboot
hoobs // to start it

**Comment by Slartybart** [comment]
Solved??

OK.... So from start to finish (on my second machine) I did the following after logging in using terminal (ssh hoobs@whatever the ip address of your machine is), password hoobsadmin.  then I typed sudo su to change user to 'root'.  This gets round a permissions error.  then:

sudo npm install -g n

sudo n stable

Then it updated node but wouldn't restart the web browser so then I did the following:

sudo npm uninstall -g @hoobs/hoobs

sudo npm install -g --unsafe-perm @hoobs/hoobs

And then it restarted with the new version of node and I was able to log back in.  Homebridge seems to be working and everything else is back to normal.   Thanks for the pointers above.
